ReactSkia/views/common/RSkDrawUtils.cpp

```diff
@@ -7,111 +7,189 @@
-
+    paint->setStyle(style);
+    paint->setStrokeWidth(strokeWidth);
+    if((borderStyle == BorderStyle::Dashed )|| (borderStyle == BorderStyle::Dotted)){
+        setPathEffect(borderStyle,strokeWidth,paint);
+    }
+}
+void setColor(SharedColor Color,Float opacity,SkPaint *paint)
+{
+    float ratio = 255.9999;
+    paint->setAntiAlias(true);
+    if(Color){
+        auto colorValue=colorComponentsFromColor(Color);
+        paint->setColor(SkColorSetARGB(
+            colorValue.alpha * ratio,
+            colorValue.red * ratio,
+            colorValue.green * ratio,
+            colorValue.blue * ratio));
     }else{
-         /*Draw Right Side*/
-         if((backgroundColor != borderProps.borderColors.right) && \
-                 (isDrawVisible(borderProps.borderColors.right,opacity) )){
-             drawEdges(RightEdge,canvas,frame,borderProps,backgroundColor,opacity);
-         }
-         /*Draw Left Side*/
-         if((backgroundColor != borderProps.borderColors.left) && \
-                 (isDrawVisible(borderProps.borderColors.left,opacity))){
-             drawEdges(LeftEdge,canvas,frame,borderProps,backgroundColor,opacity);
-         }
-         /*Draw Top Side*/
-         if((backgroundColor != borderProps.borderColors.top) && \
-                 (isDrawVisible(borderProps.borderColors.top,opacity) )){
-             drawEdges(TopEdge,canvas,frame,borderProps,backgroundColor,opacity);
-         }
-         /*Draw Bottom Side*/
-         if((backgroundColor != borderProps.borderColors.bottom) && \
-                 (isDrawVisible(borderProps.borderColors.bottom,opacity))){
-             drawEdges(BottomEdge,canvas,frame,borderProps,backgroundColor,opacity);
-         }
+        paint->setColor(DEFAULT_COLOUR);
     }
+    paint->setAlphaf((opacity >1.0 ? 1.0:opacity));
 }
+bool isDrawVisible(SharedColor Color,Float opacity,Float thickness=1.0)
+{
+    return (Color !=clearColor() && opacity >0.0 && thickness > 0.0)? true:false;
+}
+bool hasUniformBorderEdges(BorderMetrics borderProps)
+{
+    return  ( borderProps.borderColors.isUniform() &&  borderProps.borderWidths.isUniform());
+}
+void createEdge(PathMetrics pathMetrics,BorderEdges borderEdge,SkPath* path)
+{
+/*Considered the Path construct requested needs to be constructed in clockwise direction*/
 
+    bool verticalEdge=(((borderEdge == LeftEdge) || (borderEdge == RightEdge)) ? true : false);
+    bool growCW=(((borderEdge == TopEdge) ||(borderEdge == RightEdge)) ? true : false);
+    bool cornerOnRightEdge=(((borderEdge == BottomEdge) ||(borderEdge== RightEdge)||(borderEdge == TopEdge)) ? true : false);
+    bool cornerOnBottomEdge=(((borderEdge == LeftEdge) ||(borderEdge == BottomEdge ) ||(borderEdge == RightEdge)) ? true : false);
+
+    auto outterSweepAngle =(growCW ? 45 : -45);
+    auto innerSweepAngle= (growCW ? -45 : +45);;
+
+    auto outterStartX= pathMetrics.outterStart.x ;
```

Comment:
as an optimization ;than using two variables to store the x & y values can we use Point for having the x & y values 
Ex : outterStart = Point {pathMetrics.outterStart.x , pathMetrics.outterStart.y}

This way we get to manage only 4 variables here.

Reply:
Good Point . Will address it.